scripts/tasks_retrival/gene_disease_association_task_creation.py

```diff
@@ -0,0 +1,260 @@
+ Nucleic Acids Research, Volume 51, Issue D1,
+ 6 January 2023, Pages D1353–D1359,
+ https://doi.org/10.1093/nar/gkac1046
+
+
+Taken from:
+    https://platform-docs.opentargets.org/citation#latest-publication
+
+"""
+
+from pathlib import Path
+from urllib.parse import urlparse
+
+import click
+import mygene
+import pandas as pd
+
+DATA_URL = (
+    "https://ftp.ebi.ac.uk/pub/databases/opentargets/platform/23.09/output/etl/parquet/"
+)
+TASK_NAME = "Gene Disease Association"
+COLUMN_OF_SYMBOLS = "targetId"
+COLUMN_OF_OUTCOME = "score"
+
+
+def verify_source_of_data(input_file: str | None, allow_downloads: bool = False) -> str:
+    """
+    verify or provide source for data.  Data may be a local file, or if --allow-downloads is on, it will be
+    the DATA_URL.
+    This method will exit with an error if the input is not consistent with the workflow.
+
+    Args:
+    ----
+        input_file (str | None): name if input file.  None if not set, non-url path if set.
+        allow_downloads (bool, optional): has the user opted in to download the data from the source.
+            Defaults to False.
+
+    Returns:
+    -------
+        str: path to data file, wither local of the default.
+
+    """
+    if input_file is None:
+        if not allow_downloads:
+            raise ValueError(
+                f"Please enter path to local file via --input-file or turn on --allow-downloads to download task source from {input_file}"
+            )
+        # input file not given, allow download on.
+        return DATA_URL
+    elif allow_downloads:
+        raise ValueError(
+            "Arguments ambiguous:  Either give a local path of download from the web."
+        )
+    parsed_path = urlparse(str(input_file))
+    if not parsed_path.netloc == "":
+        raise ValueError(
+            f'Input path "{input_file}" is not a local file.  Please enter pre-downloaded file path or allow download'
+        )
+    return input_file
+
+
+def get_symbols(gene_targetId_list):
+    """
+        given s list of gene id's (names Like ENSG00000006468) this method
+        uses the MyGenInfo package to retrieve the gene symbol (name like PLAC4).
+
+    Args:
+    ----
+        gene_targetId_list (list): list of gene id's (names Like ENSG00000006468)
+
+    Returns:
+    -------
+        list: List of corresponding symbols
+
+    """
+    mg = mygene.MyGeneInfo()
+    list_of_gene_metadata = mg.querymany(
+        gene_targetId_list.drop_duplicates(), species="human", fields="symbol"
+    )
+    gene_metadata_df = get_id_to_symbol_df(list_of_gene_metadata)
+    return pd.DataFrame(
+        [gene_metadata_df.loc[x, "symbol"] for x in gene_targetId_list],
+        columns=["symbol"],
+    )
+
+
+def get_id_to_symbol_df(list_of_gene_metadata):
+    """
+        The method converts a list of gene metadata into a data frame,
+        each dictionary will contain the field symbol and the gene id as the query value
```

Comment:
The documentation suggests that a dictionary will be returned but the code returns a data frame.
Here we actually want a dictionary because we want a one to one data structure, , so I think returning a dictionary might be better.

Reply:
This is actually duplicate code - maybe we should be using the NCBIDescriptor to do this?

Reply:
The descriptors do not convert from ensemble to id.

Reply:
right.  we may want to create a descriptor of this type as some stage.

One way or the other, it's quite close to the code there, so I vote to keep it as is.